### PR TITLE
Linter using Github Actions

### PR DESCRIPTION
**Pull request by SarveshLimaye**
#3 

Added super-linter using github actions

### CODE REVIEW

**Review by dduportal — CHANGES_REQUESTED** [review]
Hello @SarveshLimaye thanks for this PR. 

- The current workflow succeeded, but did not executed any lint on the helm charts provided in the repository. Usually, the tool `ct` (https://github.com/helm/chart-testing) is used for linting helm charts
- Whatever tool is used, could you add a documentation to explain for a contributor how they could execute the same linting on their own machine when contributing? The idea is that we should not wait for a PR with the GitHub workflow execution to be able to lint.

**Comment by SarveshLimaye** [comment]
@dduportal Thanks for the feedback. :)
The documentation for linting has to be added in readme right?  Can I attach the link to the official doc of the super-linter? It is explained in detail there.

**Comment by lemeurherve** [comment]
>     * The current workflow succeeded, but did not executed any lint on the helm charts provided in the repository. Usually, the tool `ct` ([helm/chart-testing](https://github.com/helm/chart-testing)) is used for linting helm charts

Official GHA: https://github.com/helm/chart-testing-action

**Comment by SarveshLimaye** [comment]
> > ```
> > * The current workflow succeeded, but did not executed any lint on the helm charts provided in the repository. Usually, the tool `ct` ([helm/chart-testing](https://github.com/helm/chart-testing)) is used for linting helm charts
> > ```
> 
> Official GHA: https://github.com/helm/chart-testing-action

@lemeurherve Thanks for this . It should be exactly same as the example workflow in official gha right ?

**Comment by lemeurherve** [comment]
> @lemeurherve Thanks for this . It should be exactly same as the example workflow in official gha right ?

It should be quite similar, ot sure if it should be exactly the same.

**Comment by SarveshLimaye** [comment]
@lemeurherve @dduportal is it fine now ?  I am sorry by mistakenly I opened a new pull request. I have closed it now.

**Comment by dduportal** [comment]
Hello @SarveshLimaye , thanks for your work! The check linter is failing with the following error message:

```
Run changed=$(ct list-changed)
  changed=$(ct list-changed)
  if [[ -n "$changed" ]]; then
    echo "::set-output name=changed::true"
  fi
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.7.12/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.7.12/x64/lib
    CT_CONFIG_DIR: /opt/hostedtoolcache/ct/v3.4.0/x86_64/etc
    VIRTUAL_ENV: /opt/hostedtoolcache/ct/v3.4.0/x86_64/venv
Error: Error running process: exit status 128
Error: Process completed with exit code 1.
```

I'm not sure why to be honest... We'll take time to review next week, but if you know what to do with this error, please proceed

**Comment by SarveshLimaye** [comment]
Hello, I am not sure about the error. Can you pls review the  changes.

**Comment by lemeurherve** [comment]
@SarveshLimaye sorry for the late response, I think the problem comes from here: https://github.com/helm/chart-testing/issues/330
I'll add a suggestion to try fixing it.

File: .github/workflows/lint-test.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+on: pull_request
+
+jobs:
+  lint-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (lemeurherve) [inline .github/workflows/lint-test.yml:12]:
```suggestion
```

**Review by lemeurherve — CHANGES_REQUESTED** [review]

File: .github/workflows/lint-test.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Lint and Test Charts
+
+on: pull_request
+
+jobs:
+  lint-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (lemeurherve) [inline .github/workflows/lint-test.yml:10]:
```suggestion
        uses: actions/checkout@v2
        with:
          fetch-depth: 0
```

**Review by lemeurherve — CHANGES_REQUESTED** [review]

**Comment by lemeurherve** [comment]
Tried to remove the shallow cloning, it didn't worked...

**Comment by lemeurherve** [comment]
Related issue: https://github.com/helm/chart-testing-action/issues/69

File: .github/workflows/lint-test.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          python-version: 3.7
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.1.0
+
+      - name: Run chart-testing (list-changed)
+        id: list-changed
+        run: |
+          changed=$(ct list-changed)
```
Comment (lemeurherve) [inline .github/workflows/lint-test.yml:29]:
```suggestion
          changed=$(ct list-changed --target-branch main)
```

**Review by lemeurherve — CHANGES_REQUESTED** [review]
This should fix the check, I've made a PR in the action repository too: https://github.com/helm/chart-testing-action/pull/75

**Comment by SarveshLimaye** [comment]
> This should fix the check, I've made a PR in the action repository too: [helm/chart-testing-action#75](https://github.com/helm/chart-testing-action/pull/75)

Thank you so much for the fix . Is it fine now ?

**Comment by SarveshLimaye** [comment]
Don't know why but the test is still failing.

**Comment by dduportal** [comment]
Hi @SarveshLimaye ! Many thanks for your patience and efforts to make this happen!

I'm trying to reproduce the issue loclly but I got a bad feeling about `ct`.
I don't really understand why the "latest" release is 2.5.0 (on both the GitHub repository and homebrew on my macOS workstation), while the github action uses the 3.4.0.

Currently, the github actions reports the following error when running the command `ct lint` (with ct v3.4.0):

```shell
Linting charts...
------------------------------------------------------------------------------------------------------------------------
Error: Error linting charts: Error identifying charts to process: Error running process: exit status 128
No chart changes detected.
------------------------------------------------------------------------------------------------------------------------
Error linting charts: Error identifying charts to process: Error running process: exit status 128
```

If I try to reproduce on my current local setup (ct 2.5.0) I also got an error:

```shell
ct lint
Linting charts...
Error loading configuration: 'chart_schema.yaml' neither specified nor found in default locations
```

=> I assume that we should target 3.4.0. The goal for you and I is to reproduce the error locally before going forward.
=> Another task for you: do not forget to rebase your PR against the latest version:
* Add a git remote named `upstream` on your working directory, that point to the upstream (e.g. `git remote add upstream https://github.com/jenkins-infra/helm-charts/`, then check with `git remote -v` that the setup is ok. Once OK, or if already done, retrieve the latest changes with `git fetch --all --prune`)
** Then run the `git rebase upstream/main` command, followed by `git push origin main --force` if the rebase was successful.
** OR if you prefer merging, you can `git merge upstream/main` and `git push origin main`

**Comment by dduportal** [comment]
@SarveshLimaye Another tip: https://github.com/helm/chart-testing/blob/main/doc/ct_lint.md#synopsis .
The command `ct lint` only check for changed charts (which is an empty list by default). You need more flags to ensure that it does lint all the available charts.

**Comment by SarveshLimaye** [comment]
Hey @dduportal Thanks for the review and tips.  Will try to reproduce the error locally for 3.4.0.

**Comment by dduportal** [comment]
Hello 👋 

As the jenkins-infra team is trying to move as much workload to Jenkins controller as possible, we are closing this issue.

**Pull request closed without merging**